### PR TITLE
HPCC-13534 Improve support for distributed streaming functions

### PR DESCRIPTION
**Pull request by ghalliday**

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please review
Previous pull request: https://github.com/hpcc-systems/HPCC-Platform/pull/7306

**Comment by AttilaVamos** [comment]
Automated Smoketest Build: success
HPCC Start: OK

| setup | hthor | thor | roxie |
| ----- | ----- | ----- | ---- |
| setup | Pass | Pass | Pass |
| setup_fetch | Pass | Pass | Pass |
| setupdict | Pass | Pass | Pass |
| setupsearchindex | Pass | Pass | Pass |
| setupsq | Pass | Pass | Pass |
| setupwordindex | Pass | Pass | Pass |
| setupxml | Pass | Pass | Pass |
 

| test | hthor | thor | roxie |
| ----- | ----- | ----- | ---- |
| streamread | Pass | Pass | Pass |
 
HPCC Stop: OK

File: ecl/hqlcpp/hqlhtcpp.cpp [hunk 1]
```diff
@@ -18672,6 +18690,13 @@ static bool needsRealThor(IHqlExpression *expr, unsigned flags)
     case no_extractresult:
         return needsRealThor(expr->queryChild(0), flags);
 
+    case no_call:
+    case no_externalcall:
+        if (isDistributedFunctionCall(expr))
+            return true;
+        //MORE: check for streamed inputs.
```
Comment (richardkchapman) [inline ecl/hqlcpp/hqlhtcpp.cpp]: Is there a JIRA open for the additional work?

**Comment by richardkchapman** [comment]
One minor question. @jakesmith are you still happy with the thor change?

**Comment by jakesmith** [comment]
Yep, looks good.
@richardkchapman

File: ecl/hqlcpp/hqlhtcpp.cpp (hunk 1 above)
Comment (ghalliday) [inline ecl/hqlcpp/hqlhtcpp.cpp]: Yes - See HPCC-12921 and child tasks.

**Pull request merged**